lottie-ios/Classes/Private/LOTAnimationView.m

```diff
@@ -208,14 +210,32 @@ - (CGFloat)_progressForFrame:(NSNumber *)frame {
   if (!_sceneModel) {
     return 0;
   }
-  return ((frame.floatValue - _sceneModel.startFrame.floatValue) / (_sceneModel.endFrame.floatValue - _sceneModel.startFrame.floatValue));
+  CGFloat absoluteProgress = ((frame.floatValue - _sceneModel.startFrame.floatValue) / (_sceneModel.endFrame.floatValue - _sceneModel.startFrame.floatValue));
+  if ([self _isPlayingForwards]) {
+    return absoluteProgress;
+  } else {
+    // If the animation is playing backwards, the progress is inverted.
+    return 1 - absoluteProgress;
```

Comment:
_animationProgress should always be in absolute time, not relative time. So if an animation is playing in reverse the progress will decrease, if it plays in autoReverse it will go up then down.